### PR TITLE
Be Excel-compatible when handling pre 1900-03-01 dates

### LINKED ISSUE #145: Handling dates before 1900-03-01
Hi,

It seems even though 1900 is not a leap year, it is treated as such in Excel (see https://en.wikipedia.org/wiki/Microsoft_Excel#Fictional_leap_day_in_the_year_1900). The wikipedia page also mentions this has been carried over to the Open XML file format.

This means 1900-02-29 is a valid date in Excel, and dates before 1900-03-01 are all shifted one day ahead when they're generated by xlsx. I'm not sure what the current behaviour is when converting from Excel.

Do I understand this correctly and is it a bug in xlsx?

### PR DESCRIPTION
**Pull request by MarceloZabini**
This hopefully fixes #145.
Although this should bring us closer to Excel, it does break compatibility, so it's probably worth mentioning in the changelog?

Thanks for your work and please let me know any changes you might want!

### CODE REVIEW

**Comment by MarceloZabini** [comment]
I forgot to attach a screenshot of Online Excel that may or may not help reviewing. Cells in the same row are equal but with different formatting:

image

**Review by qrilka — COMMENTED** [review]
Having a changelog entry makes total sense.
As for the PR itself - only a minor comment about haddocks, otherwise LGTM.

File: src/Codec/Xlsx/Types/Common.hs [hunk 1]
```diff
@@ -251,23 +252,41 @@ baseDate :: DateBase -> Day
...
 
 -- | Converts datetime into serial value
 dateToNumber :: Fractional a => DateBase -> UTCTime -> a
 dateToNumber b (UTCTime day diffTime) = numberOfDays + fractionOfOneDay
   where
-    numberOfDays = fromIntegral (diffDays day $ baseDate b)
+    numberOfDays = fromIntegral (diffDays excel1900CorrectedDay $ baseDate b)
     fractionOfOneDay = realToFrac diffTime / (24 * 60 * 60)
+    -- For historical reasons, Excel thinks 1900 is a leap year (https://docs.microsoft.com/en-gb/office/troubleshoot/excel/wrongly-assumes-1900-is-leap-year).
+    -- So dates strictly before March 1st 1900 need to be corrected.
```
[the remark this answers is not in the document]
Comment (MarceloZabini) [reply src/Codec/Xlsx/Types/Common.hs]: Good point. I've changed the haddocks to contain this link and some more info as well, and noticed that this new behaviour makes `dateFromNumber` and `dateToNumber` no longer inverses of each other under certain circumstances, which I also documented.

**Comment by MarceloZabini** [comment]
I added an entry in the changelog but I don't know how you want to handle versioning, so feel free to change, suggest a change or just ask me to change something there.

**Comment by qrilka** [comment]
Sorry, was a bit busy - I plan to take a look into it this evening

**Comment by qrilka** [comment]
I think your change makes sense, I want to release a new version with #144 merged so could change the version but that's not done yet.

**Pull request merged**